File: problems/Owl_race/submissions/accepted/owl_race.py
def owl_race(x1, v1, x2, v2):
    if (v1 == v2):
        return "YES" if(x1 == x2) else "NO"
    else:
        t = int(abs((x2-x1)/(v1-v2)))
        eq1 = x1 + v1 * t
        eq2 = x2 + v2 * t
        if (eq1 == eq2):
            return "YES"
        else:
            return "NO"


x1, v1, x2, v2 = map(int, input().split())

print(owl_race(x1, v1, x2, v2))

File: problems/Owl_race/submissions/accepted/owl_race.java
import java.util.Scanner;
import java.lang.Math;

public class owl_race {
    public static void main(String[] args) {
        int x1, v1, x2, v2;
        Scanner sc = new Scanner(System.in);
        x1 = sc.nextInt();
        v1 = sc.nextInt();
        x2 = sc.nextInt();
        v2 = sc.nextInt();

        if (v1 == v2) {
            if (x1 == x2)
                System.out.print("YES");
            else
                System.out.print("NO");
        } else {
            int t = Math.abs((x2 - x1) / (v1 - v2));
            int eq1 = x1 + v1 * t;
            int eq2 = x2 + v2 * t;
            if (eq1 == eq2)
                System.out.print("YES");
            else
                System.out.print("NO");
        }
        sc.close();
    }
}